gcp-observability/src/main/proto/grpc/observabilitylog/v1/observabilitylog.proto

```diff
@@ -28,151 +28,99 @@ option java_outer_classname = "ObservabilityLogProto";
-  // from an unset value.
-  // Each call may have several log entries. They will all have the same rpc_id.
+  // Uniquely identifies a call.
+  // Each call may have several log entries. They will all have the same call_id.
   // Nothing is guaranteed about their value other than they are unique across
   // different RPCs in the same gRPC process.
-  string rpc_id = 2;
+  string call_id = 2;
 
-  EventType event_type = 3;      // one of the above EventType enum
-  EventLogger event_logger = 4;  // one of the above EventLogger enum
+  // The entry sequence ID for this call. The first message has a value of 1,
+  // to disambiguate from an unset value. The purpose of this field is to
+  // detect missing entries in environments where durability or ordering is
+  // not guaranteed.
+  uint64 sequence_id = 3;
 
-  // the name of the service
-  string service_name = 5;
-  // the name of the RPC method
-  string method_name = 6;
+  EventType type = 4;      // one of the above EventType enum
+  EventLogger logger = 5;  // one of the above EventLogger enum
 
-  LogLevel log_level = 7;        // one of the above LogLevel enum
+  // Payload for log entry.
+  // It can include a combination of {metadata, message, status based on type of
+  // the event event being logged and config options.
+  Payload payload = 6;
 
+  // true if message or metadata field is either truncated or omitted due
+  // to config options
+  bool payload_truncated = 7;
   // Peer address information. On client side, peer is logged on server
   // header event or trailer event (if trailer-only). On server side, peer
   // is always logged on the client header event.
-  Address peer_address = 8;
-
-  // the RPC timeout value
-  google.protobuf.Duration timeout = 11;
+  Address peer = 8;
 
   // A single process may be used to run multiple virtual servers with
   // different identities.
   // The authority is the name of such a server identify. It is typically a
   // portion of the URI in the form of <host> or <host>:<port>.
-  string authority = 12;
-
-  // Size of the message or metadata, depending on the event type,
-  // regardless of whether the full message or metadata is being logged
-  // (i.e. could be truncated or omitted).
-  uint32 payload_size = 13;
-
-  // true if message or metadata field is either truncated or omitted due
-  // to config options
-  bool payload_truncated = 14;
-
-  // Used by header event or trailer event
-  Metadata metadata = 15;
-
-  // The entry sequence ID for this call. The first message has a value of 1,
-  // to disambiguate from an unset value. The purpose of this field is to
-  // detect missing entries in environments where durability or ordering is
-  // not guaranteed.
-  uint64 sequence_id = 16;
-
-  // Used by message event
-  bytes message = 17;
+  string authority = 10;
+  // the name of the service
+  string service_name = 11;
+  // the name of the RPC method
+  string method_name = 12;
+}
 
+message Payload {
```

Comment:
based on how Payload is used, metadata, message and status will never be used at the same time since all those events are separate. How about using oneof in that case?

(I understand this is from the design so I have added a comment there as well)